src/Core/Table/ArcTable.fs

```diff
@@ -372,42 +391,49 @@ type ArcTable(name: string, headers: ResizeArray<CompositeHeader>, values: Syste
 
     // - Column API - //
     // GetColumnAt?
-    member this.GetColumn(columnIndex:int) : CompositeColumn =
+    member this.GetColumn(columnIndex:int, ?fillDefault : bool) : CompositeColumn =
+        let fillDefault = defaultArg fillDefault false
         SanityChecks.validateColumnIndex columnIndex this.ColumnCount false
         let h = this.Headers.[columnIndex]
-        let cells = [|
-            for i = 0 to this.RowCount - 1 do 
-                match this.TryGetCellAt(columnIndex, i) with
-                | None -> failwithf "Unable to find cell for index: (%i, %i)" columnIndex i
-                | Some c -> c
-        |]
+        let cells =
+            [|
```

Comment:
This can be optimized by first checking if columnIndex points to a `Constant` columnRef, then we can just fill with the same value